### DISCUSSION TITLE
In Demo, how to set order ID sequentially

### DISCUSSION THREAD

**Discussion by emincer** [filamentphp/filament#6627] [Help]
So I am doing something vaguely similar to the demo. In the demo, you use two steps and the first step sets an Order ID to "OR-" plus a random number.

In my use case, I am creating a template with many template_items so I am using a repeater like you do in the demo. But it's all one step and I would like to create a new template with the next sequential id, and store all the template_items with it. Since I have not set the template id naturally the error I'm getting is 
`SQLSTATE[HY000]: General error: 1364 Field 'template_id' doesn't have a default value`
<img width="1345" alt="Screenshot 2023-05-28 at 2 01 55 PM" src="https://github.com/filamentphp/filament/assets/1903132/65b978bc-f761-41cb-9d3d-04cbbe0ed685">

What's the proper way to do this?

**Comment by emincer** [comment]
And to elaborate I guess I'm confused how create does the save with the hasMany relationship. Clearly the template needs to be saved first so I have a record id that the template_items can each use with a template_id. Sorry, this could be Laravel n00b question, again, but I'm just not seeing how this magic happens.

**Comment by ArtMin96** [comment]
Add `template_id` in model `$fillables` array

**Reply by emincer** [reply]
That was super helpful, thanks!

Although now I get 
    `SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '23' for key 'newtemplateitems.PRIMARY'`

Which is strange because, being a repeater, I thought it would add each `newtemplateitem` as a new row. Also, it is not setting the `template_id` at all, it's just NULL.
